### PR TITLE
Drop python3.7 support. Refactoring

### PR DESCRIPTION
**Pull request by babenek**
## Description

- skip Python3.7 support due [pandas source does not support it](https://pandas.pydata.org/pandas-docs/stable/getting_started/install.html#python-version-support) 
- added used package to requirements explicitly 
- refactoring to reduce cyclomatic complexity
- prepare for support binary diffs after https://github.com/cscorley/whatthepatch/pull/53

## How has this been tested?

Please describe the tests that you ran to verify your changes.

- [x] UnitTests
- [x] Benchmark

### CODE REVIEW

File: credsweeper/utils/util.py [hunk 1]
```diff
@@ -372,6 +375,30 @@ def yaml_dump(obj: Any, file_path: str, encoding=DEFAULT_ENCODING) -> None:
         except Exception as exc:
             logging.error(f"Failed to write: {file_path} {exc}")
 
+    @staticmethod
+    def __extract_value_value(node: Any, value_value: Any) -> List[Any]:
```
Comment (csh519) [inline credsweeper/utils/util.py:379]: How about rename the method with `__extract_value()`?
IMO `value_value` seems not that good..

**Review by csh519 — APPROVED** [review]
Python 3.7 is dropped well and code has been refactored.

LGTM :+1:

File: credsweeper/file_handler/data_content_provider.py [hunk 2]
```diff
@@ -46,6 +45,11 @@ def data(self, data: bytes) -> None:
         """data setter"""
         self.__data = data
 
+    def __is_structure(self) -> bool:
```
Comment (kmnls) [inline credsweeper/file_handler/data_content_provider.py:48]: __has_structure ? __is_structured_data ? __format_recognized ?

Comment (babenek) [reply credsweeper/file_handler/data_content_provider.py:48]: too complicated question for the private member name :)

**Review by kmnls — APPROVED** [review]
Update is good except of some bad-formed names related to the structured data processing,

**Pull request merged**